modules/meshrenderinggl/include/modules/meshrenderinggl/datastructures/transformedrasterization.h

```diff
@@ -0,0 +1,74 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <modules/meshrenderinggl/meshrenderingglmoduledefine.h>
+#include <modules/meshrenderinggl/datastructures/rasterization.h>
+
+namespace inviwo {
+/**
+ * \brief Rasterization wrapper to add an additional transform on the render call.
+ */
+class IVW_MODULE_MESHRENDERINGGL_API TransformedRasterization : public Rasterization {
+public:
+    TransformedRasterization(std::shared_ptr<const Rasterization> rasterization,
+                             const mat4& additionalWorldTransform)
+        : innerRasterization_(rasterization), additionalWorldTransform_(additionalWorldTransform) {}
+
+    virtual void rasterize(const ivec2& imageSize, const mat4& worldMatrixTransform,
+                           std::function<void(Shader&)> setUniforms) const override {
+        innerRasterization_->rasterize(imageSize, additionalWorldTransform_ * worldMatrixTransform,
+                                       setUniforms);
+    }
+    virtual bool usesFragmentLists() const override {
+        return innerRasterization_->usesFragmentLists();
+    }
+
+    virtual Document getInfo() const override { return innerRasterization_->getInfo(); }
+    virtual Rasterization* clone() const override { return new TransformedRasterization(*this); }
+
+protected:
+    std::shared_ptr<const Rasterization> innerRasterization_;
+    const mat4 additionalWorldTransform_;
+};
+
+/**
+ * \brief A very simple SpatialEntity<3> to handle a world and model transform
+ * Used by rasterizations for adding transforms before rendering, without copying the mesh data.
+ */
+struct IVW_MODULE_MESHRENDERINGGL_API CompositeTransform : public SpatialEntity<3> {
```

Comment:
Why not just use SpatialEntity directly? this class does nothing now? or did i miss something?

Reply:
Or simply just a matrix?

Reply:
This is used to add all the uniforms in the shaders in a convenient and standard way...
We also add the inverse etc...

Reply:
SpatialEntity is a purely virtual class, so can't do. This is the most lightweight way of doing that, I suppose.

Reply:
We could of course make SpatialEntity instantiable by just setting a clone implementation on the base class. I think I prefer this explicit way though.

Reply:
ahh ok